lib/server.js

```diff
@@ -1366,6 +1369,22 @@ Server.prototype._finishReqResCycle = function _finishReqResCycle(
         var finalErr = err || res.err;
         req.emit('restifyDone', route, finalErr);
         self.emit('after', req, res, route, finalErr);
+    } else if (
+        res._handlersFinished === true &&
+        res.headersSent === false &&
+        !res.err
+    ) {
+        // if we reached the end of the handler chain and headers haven't been
+        // sent AND there isn't an existing res.err (e.g., req abort/close),
+        // it's possible it's a user error and a response was never written.
+        // send a 500.
+        res.send(
+            new errors.InternalServerError(
+                'reached the end of the handler chain without ' +
```

Comment:
Doesn't this get sent back to the client as:

```
{ "code": "Internal Server Error", "message": "reached the end of the handler chain without writing a response!" }
```

Have we thought about how much this reveals about the server?

Reply:
Good point. I don't feel super strongly about this, and its worth noting we already send back pretty specific error messages for various other scenarios (bad redirects, bad formatters, etc). The messages are scrubbable using the error events API but makes me think we could do a better job w/ standardized error codes for restify.

Reply:
I don't think it reveals anything that shouldn't be fixed. Maybe a more generic message like "request finished without response", or "sever didn't send response" is enough.